lms/static/js/spec/search/search_spec.js

```diff
@@ -0,0 +1,489 @@
+            var term = '  search string  ';
+            $('.search-field').val(term);
+            this.form.doSearch(term);
+            expect(this.onSearch).toHaveBeenCalledWith($.trim(term));
+            expect($('.search-field').val()).toEqual(term);
+            expect($('.search-field')).toHaveClass('is-active');
+            expect($('.search-button')).toBeHidden();
+            expect($('.cancel-button')).toBeVisible();
+        });
+
+        it('triggers a search event and changes to active state', function () {
+            var term = 'search string';
+            $('.search-field').val(term);
+            $('form').trigger('submit');
+            expect(this.onSearch).toHaveBeenCalledWith(term);
+            expect($('.search-field')).toHaveClass('is-active');
+            expect($('.search-button')).toBeHidden();
+            expect($('.cancel-button')).toBeVisible();
+        });
+
+        it('clears search when clicking on cancel button', function () {
+            $('.search-field').val('search string');
+            $('.cancel-button').trigger('click');
+            expect($('.search-field')).not.toHaveClass('is-active');
+            expect($('.search-button')).toBeVisible();
+            expect($('.cancel-button')).toBeHidden();
+            expect($('.search-field')).toHaveValue('');
+        });
+
+        it('clears search when search box is empty', function() {
+            $('.search-field').val('');
+            $('form').trigger('submit');
+            expect(this.onClear).toHaveBeenCalled();
+            expect($('.search-field')).not.toHaveClass('is-active');
+            expect($('.cancel-button')).toBeHidden();
+            expect($('.search-button')).toBeVisible();
+        });
+
+    });
+
+
+    describe('SearchItemView', function () {
+
+        beforeEach(function () {
+            TemplateHelpers.installTemplate('templates/courseware_search/search_item');
+            this.model = {
+                attributes: {
+                    location: ['section', 'subsection', 'unit'],
+                    content_type: 'Video',
+                    excerpt: 'A short excerpt.',
+                    url: 'path/to/content'
+                }
+            };
+            this.item = new SearchItemView({ model: this.model });
+        });
+
+        it('has useful html attributes', function () {
+            expect(this.item.$el).toHaveAttr('role', 'region');
+            expect(this.item.$el).toHaveAttr('aria-label', 'search result');
+        });
+
+        it('renders underscore template', function () {
+            var href = this.model.attributes.url;
+            var breadcrumbs = 'section ▸ subsection ▸ unit';
+
+            this.item.render();
+            expect(this.item.$el).toContainHtml(this.model.attributes.content_type);
+            expect(this.item.$el).toContainHtml(this.model.attributes.excerpt);
+            expect(this.item.$el.find('a[href="'+href+'"]')).toHaveAttr('href', href);
```

Comment:
If you do a find with an href in the selector, then won't it by definition have the attr? Maybe you want to just verify that a result is returned?